include/envoy/network/transport_socket.h

```diff
@@ -149,9 +149,12 @@ class TransportSocketFactory {
   virtual bool implementsSecureTransport() const PURE;
 
   /**
+   * @param overrideServerNameIndication set SNI, disregard the value the factory was configured
+   * with
    * @return Network::TransportSocketPtr a transport socket to be passed to connection.
    */
-  virtual TransportSocketPtr createTransportSocket() const PURE;
+  virtual TransportSocketPtr
+  createTransportSocket(std::string overrideServerNameIndication = "") const PURE;
```

Comment:
no default parameter in virtual function:
https://google.github.io/styleguide/cppguide.html#Default_Arguments

you may want this as absl::optional<std::string>